### PR TITLE
tests_gaudi: Added L2 vllm workload

### PR DESCRIPTION
**Pull request by vbedida79**
PR includes gaudi l2 vllm workload

- buildconfig docker image is based on PR https://github.com/HabanaAI/vllm-fork/pull/602
- deployment including secret for hugging face token and pvc for model cache 
- Readme steps to build, deploy and verify

Signed-off-by: vbedida79 <veenadhari.bedida@intel.com>

### CODE REVIEW

File: tests/gaudi/l2/vllm_buildconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  name: vllm-workload
+  namespace: gaudi-validation
+spec:
+  triggers:
+    - type: "ConfigChange"
+    - type: "ImageChange"
+  runPolicy: "Serial"
+  source:
+    git:
+      uri: https://github.com/opendatahub-io/vllm.git
```
Comment (uMartinXu) [inline tests/gaudi/l2/vllm_buildconfig.yaml:23]: After comparing
1. https://github.com/opendatahub-io/vllm.git - ODH fork vllM
2. https://github.com/vllm-project/vllm - vLLM upstream
3.https://github.com/HabanaAI/vllm-fork - Habana fork vLLM
I think currently we should start from use the 3. with the change in 1 (adding the ubi based docker file for RH OpenShift), and obviously the Intel are upstreaming from 3 to 2. So in the long run we will using 2. 
So I think we need to 1). submit a PR to adding the ubi based docker file for RH, and also add the RH 9.4 support into the documents, and then 2). using repo 3   3) I think the owner of 3 will also help to upstream the ubi based docker file and doc to 2. 4) after that we can switch to use 2 the upstream vLLM.
@vbedida79  any comments? :-)

**Comment by vbedida79** [comment]
Updated according to comments, please review. thanks

File: tests/gaudi/l2/vllm_buildconfig.yaml (hunk 1 above)
Comment (vbedida79) [reply tests/gaudi/l2/vllm_buildconfig.yaml:23]: https://github.com/HabanaAI/vllm-fork/pull/190 PR for ubi from RH can be merged into vllm gaudi repo, we can use that directly. Currently we can use the ubi image by RH maintained in https://github.com/opendatahub-io/vllm.git from https://github.com/HabanaAI/vllm-fork

File: tests/gaudi/l2/README.md [hunk 2]
```diff
@@ -75,3 +75,104 @@ Welcome to HCCL demo
 [BENCHMARK]     Algo Bandwidth : 147.548069 GB/s
 ####################################################################################################
 ```
+<<<<<<< HEAD
```
Comment (uMartinXu) [inline tests/gaudi/l2/README.md:78]: PR git comments: the buildconfig base on https://github.com/HabanaAI/vllm-fork/pull/602

File: tests/gaudi/l2/vllm_deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      containers:
+        - name: vllm-container
+          image: image-registry.openshift-image-registry.svc:5000/gaudi-validation/vllm-workload:latest
+          command: [ "/bin/bash", "-c", "--" ]
+          args: ["vllm serve meta-llama/Llama-3.1-8B"] # Add the model 
+          ports:
+          - containerPort: 8000
+          resources:
+            limits:
+              habana.ai/gaudi: 4
```
Comment (uMartinXu) [inline tests/gaudi/l2/vllm_deployment.yaml:42]: Could we check and confirm how many Accelerators are actually used by vLLM? 
I suggest to start from using only single Accelerator.

File: tests/gaudi/l2/README.md (hunk 2 above)
Comment (vbedida79) [reply tests/gaudi/l2/README.md:78]: will add these

File: tests/gaudi/l2/README.md [hunk 4]
```diff
@@ -75,3 +75,104 @@ Welcome to HCCL demo
...
+=======
+
+## vLLM 
+vLLM is a serving engine for LLM's. The following workloads deploys a VLLM server with an LLM using Intel Gaudi. Refer to [Intel Gaudi vLLM fork](https://github.com/HabanaAI/vllm-fork.git) for more details.
+
+Build the workload container image:
+```
+git clone https://github.com/opendatahub-io/vllm.git --branch gaudi-main
+
```
[the remark this answers is not in the document]
Comment (vbedida79) [reply tests/gaudi/l2/README.md:87]: there is no 1.18.0 branch in odh vllm repo, its called gaudi-main which is latest

File: tests/gaudi/l2/vllm_deployment.yaml (hunk 3 above)
Comment (vbedida79) [reply tests/gaudi/l2/vllm_deployment.yaml:42]: I can check for 1 resource and update

**Review by uMartinXu — APPROVED** [review]

**Pull request merged**